needy/local_configuration.py

```diff
@@ -27,25 +26,31 @@ def __enter__(self):
                 if not os.path.exists(directory):
                     raise e
 
-        self.__fd = lock_file(self.__path, timeout=0)
-        if self.__fd is None:
+        try:
+            self.__file = open(self.__path, 'r+')
+        except IOError:
+            self.__file = open(self.__path, 'w+')
```

Comment:
Really annoyed that I can't seem to do this in one call. Intuitively, it seems like there could be race conditions, but so far I haven't been able to uncover any issues by executing these with odd timing from different python shells.